testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -400,6 +400,67 @@ impl RelayerThread {
         || !self.config.miner.wait_for_block_download
     }
 
+    /// Compute and set the global IBD flag from a NetworkResult
+    pub fn infer_ibd(&mut self, net_result: &NetworkResult) {
+        let cur_sn = SortitionDB::get_canonical_burn_chain_tip(self.sortdb.conn())
+            .expect("FATAL: failed to query sortition DB");
+
+        // (remember, get_headers_height() is 1-indexed, so account for that with -1)
+        let headers_height = self
+            .bitcoin_controller
+            .get_headers_height()
+            .saturating_sub(1);
+
+        // are we still processing sortitions?
+        let burnchain_ibd = cur_sn.block_height != headers_height;
+
+        // if the highest available tenure is known, then is it the same as the ongoing stacks
+        // tenure?  If so, then we're not IBD. If not, then we're IBD.
+        // If it is not known, then we're not in IBD.
```

Comment:
Doesn't this mean that a running node operating at chain tip would switch into `IBD = true` at every tenure boundary?